File: javascript/leetcode/1431. Kids With the Greatest Number of Candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function (candies, extraCandies) {
  const ans = new Array(candies.length);

  let _max = Math.max(...candies);

  candies.forEach((el, ind) => {
    ans[ind] = el + extraCandies >= _max ? true : false;
  });

  return ans;
};

File: c++/leetcode/1431. Kids With the Greatest Number of Candies.cpp
class Solution
{
public:
  vector<bool> kidsWithCandies(vector<int> &candies, int extraCandies)
  {
    vector<bool> ans(candies.size());

    auto _max = *max_element(candies.begin(), candies.end());

    for (int i = 0; i < candies.size(); ++i)
    {
      ans[i] = candies[i] + extraCandies >= _max ? true : false;
    }

    return ans;
  }
};
